src/libcore/ptr/mod.rs

```diff
@@ -1074,17 +1074,22 @@ impl<T: ?Sized> *const T {
     /// operation because the returned value could be pointing to invalid
     /// memory.
     ///
-    /// When calling this method, you have to ensure that if the pointer is
-    /// non-NULL, then it is properly aligned, dereferencable (for the whole
-    /// size of `T`) and points to an initialized instance of `T`. This applies
-    /// even if the result of this method is unused!
+    /// When calling this method, you have to ensure that *either* the pointer is NULL *or*
+    /// all of the following is true:
+    /// - it is properly aligned
+    /// - it must point to an initialized instance of T; in particular, the pointer must be
+    ///   "dereferencable" in the sense defined [here].
+    ///
+    /// This applies even if the result of this method is unused!
     /// (The part about being initialized is not yet fully decided, but until
     /// it is, the only safe approach is to ensure that they are indeed initialized.)
     ///
     /// Additionally, the lifetime `'a` returned is arbitrarily chosen and does
-    /// not necessarily reflect the actual lifetime of the data. It is up to the
-    /// caller to ensure that for the duration of this lifetime, the memory this
-    /// pointer points to does not get written to outside of `UnsafeCell<U>`.
+    /// not necessarily reflect the actual lifetime of the data. *You* must enforce
+    /// Rust's aliasing rules. In particular, for the duration of this lifetime,
+    /// the memory the pointer points to must not get mutated (except inside `UnsafeCell`).
+    ///
+    /// [here]: ../ptr/index.html#safety
```

Comment:
the issue here is that this text appears on three pages:

* https://doc.rust-lang.org/std/ptr/index.html
* https://doc.rust-lang.org/core/ptr/index.html
* https://doc.rust-lang.org/std/primitive.pointer.html

In particular, note how the third link has one less layer of directory than the others. So your `../ptr` is messing up, because, if you convert the link on each page:

* https://doc.rust-lang.org/std/ptr/../ptr/index.html#safety
* https://doc.rust-lang.org/core/ptr/../ptr/index.html#safety
* https://doc.rust-lang.org/std/../ptr/index.html#safety

removing the `..`:

* https://doc.rust-lang.org/std//ptr/index.html#safety
* https://doc.rust-lang.org/core/ptr/index.html#safety
* https://doc.rust-lang.org/ptr/index.html#safety

that last link does not exist.

Luckily, it's not actually clear to me why you need the `../ptr` in the first place. try 

    /// [here]: index.html#safety

instead.

Reply:
> index.html#safety

This was the first option I tried, it didn't work either. 

> it's not actually clear to me why you need the `../ptr` in the first place.

I just noticed some doc comments in this file use this path so I tried it.

Reply:
Like I said, it doesn't work. Any more ideas?

Reply:
Well, looks we'll have to use absolute paths then, rustdoc does not give us a choice. Cc @GuillaumeGomez 
```
/// [here]: /std/ptr/index.html#safety
```

Reply:
Please do not use paths starting with "/". Go to the docs and add as many ".." as required.

Reply:
@GuillaumeGomez please see the context here -- it is not possible to add the right number of `../` as the same markdown is displayed on multiple pages in the docs. This is not the first time that has come up, see https://github.com/rust-lang/rust/issues/63944.

Reply:
So, have you came up with a consensus of sorts?

Reply:
Based on [this precedence case](https://github.com/rust-lang/rust/blob/d9025395c8e46599f062216c818e3388e597d553/src/libcore/slice/mod.rs#L303), I would go for using `https://doc.rust-lang.org/std/ptr/index.html#safety`.

Reply:
Oh right. Well, we can't use link like this either and the intra-doc links don't support putting a "#" in it (I'll open an issue about this). Do you mind waiting for me to allow "#" in intra doc links,

Reply:
@GuillaumeGomez Not at all, take your time.

Reply:
@GuillaumeGomez what is an "intra-doc link"?

Reply:
@RalfJung The name might be misleading but we never really spoke about it. :p

Reply:
I opened #66675 yesterday btw. So should be there in not too much time.